### ISSUE TITLE
using the emulator, upload a file then downloading it return 404

### ISSUE THREAD

**Issue by Raffaello** [oittaa/gcp-storage-emulator#209]
**Describe the bug**
uploading a file succesfully, then when try to downloading is returning 404.

```
[SERVER] All services started
"POST /upload/storage/v1/b/local-bucket/o?name=Example.xml&uploadType=resumable HTTP/1.1" 200 -
No file to remove '_resumable/6effa36d984da5caa33e4f5fb3a0f70e54ab18f801d03504f7cd2fdaa38e2cc5'
"PUT /upload/storage/v1/b/local-bucket/o?name=Example.xml&uploadType=resumable&upload_id=local-bucket%3AExample.xml%3A2022-10-21+13%3A31%3A18.531160 HTTP/1.1" 200 -
Resource not found:
**resource 'b' not found**
"GET /b/local-bucket/o/Example.xml?alt=media HTTP/1.1" 404 -
```

the url to get `/b` is the standard GCP Storage url bucket `/o` obect. 
it isn't expected to fail on a `GET` `/b` url

**To Reproduce**
[A minimal, complete, and reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) to reproduce the behavior.

1. upload a file with NodeJS
2. downaload a file with NodeJS

```js
import { Storage } from "@google-cloud/storage"
import process from "process"

const HOST = "localhost"
const PORT = 9023
const BUCKET = "local-bucket"

process.env.STORAGE_EMULATOR_HOST = `http://${HOST}:${PORT}`

const storage = new Storage()

async function upload(filePath, destinationFilename) {
    const resp = await storage.bucket(BUCKET).upload(filePath,{
        destination: destinationFilename
    })
}
// USAGE needs filepath to upload and the name of the file at destination:
// e.g. local_storage_upload mypath/filename.txt filename.txt
upload(process.argv[2], process.argv[3]).catch(console.error)

// usage npx local-storage_upload.ts filetoUpload.txt filetoUpload.txt
```
download:
```js
const [fileData] = await this.storage.bucket("local-bucket").file("fileToUpload.txt).download({ validation: "md5" });
// this will trigger 404 and throw exception as file not found
```

**Expected behavior**
it should download the file

**System (please complete the following information):**

- gcp-storage-emulator version: latest, there is no way to retrieve from -h argument or a --version argument.

**Additional context**

Basically what is saying is "url not found" `b/" the b/ part of the path is the "bucket" in GCP storage. so it should just work without any further issue as it is the basic GCP storage url

**Comment by EmandM** [comment]
Bumping because I also have this issue, however mine is failing with:
```
storage_emulator_1    | Resource not found:
storage_emulator_1    | resource 'storage' not found
storage_emulator_1    | "GET /storage/v1/b/7ca33333c2c80d7ccb7954c1b79102c960f45700ff96e9cac47f950d/iam?alt=json&optionsRequestedPolicyVersion=1&prettyPrint=false HTTP/1.1" 404 -
```

**Comment by Raffaello** [comment]
@EmandM 
if i recall it correctly i had to use an ENV parameter detection: when there is STORAGE_EMULATOR_URL or something i was changing slightly the URL i don't remember if it was for mounting the emulator or in the URL of the application . but something like that. anyway i can't help you more than this as i don't remember the workaround that i adopted completely.

**Comment by EmandM** [comment]
This doesn't solve the issue I am having, but is the solution to the original issue https://github.com/googleapis/google-cloud-go/issues/4444#issuecomment-898716815

The issue is if you use `STORAGE_EMULATOR_HOST=http://localhost:9090` the go storage api will overwrite a whole bunch of things. You have to set it as in the issue above. If you use just `STORAGE_EMULATOR_HOST=localhost:9090` (i.e. without the `http://` then it works fine.

**Comment by i-am-ian** [comment]
I ran into this issue today. There's an [open issue](https://github.com/googleapis/nodejs-storage/issues/2092) to address this in the nodejs-storage project.

In the meantime, I found a workaround - rather than setting the STORAGE_EMULATOR_HOST variable, set the apiEndpoint property in the options provided to the Storage constructor.

`const storage = new Storage({ apiEndpoint: 'http://127.0.0.1:9023' });`

That's working for both uploading and downloading a file for me. I haven't tried other operations.

**Comment by patilavisala815-commits** [comment]
HBIE<>

**Comment by oittaa** [comment]
Fixed on current `main` (released via recent streaming / path-compat work, including #325).

**Root cause:** With `STORAGE_EMULATOR_HOST`, the Node client downloads via unprefixed paths like `GET /b/<bucket>/o/<object>?alt=media`. The emulator only registered `/storage/v1/b/...`, so the public `/{bucket}/{object}` route treated `b` as the bucket name → 404 (“resource 'b' not found”).

**Fix:** JSON API handlers are registered under both `/storage/v1/b/...` and `/b/...`. Resumable streaming edge cases (empty metadata body `{}`, Node `Content-Range`) were fixed in the same release train.

**Verified:** Resumable upload then `GET /b/.../o/...?alt=media` returns 200 with correct content; Node `STORAGE_EMULATOR_HOST` createBucket → upload → download succeeds.

Please reopen with a minimal repro if anything still fails on the latest release.
